generator/lib/control.ml

```diff
@@ -46,6 +46,35 @@ let api =
           ];
         ]
       )) in
+  let configuration = {
+    Arg.name = "configuration";
+    ty = Type.(Dict(String, Basic String));
+    description = String.concat " " [
+                    "Plugin-specific configuration which describes where and";
+                    "how to locate the storage repository. This may include";
+                    "the physical block device name, a remote NFS server and";
+                    "path or an RBD storage pool.";
+                  ];
+  } in
+  let probe_result_decl =
+    Type.(Struct (
+        ( configuration.name, configuration.ty, configuration.description ),
+        [ "complete", Basic Boolean, String.concat " " [
+            "True if this configuration is complete and can be used to";
+            "call SR.create. False if it requires further iterative calls";
+            "to SR.probe, to potentially narrow down on a configuration";
+            "that can be used.";
```

Comment:
I found the name of this field a bit confusing after reading the xapi-storage-script code: We probably do have "complete" configuration, when we consider its contents, for existing SRs (when `sr` is present), but we cannot use it for creating a new SR, because there is already an SR with this configuration.

Reply:
I wonder whether the SMAPIv3 plugin itself should just return the map in its final form (if we agree that this is the final form we want):
https://github.com/xapi-project/xcp-idl/pull/208

Reply:
Good question - that also sounds good to me. Or we could change this field name if others also find it easy to misunderstand.

Reply:
Discussed with @jonludlam the xapi-storage and xcp-idl interfaces for SMAPIv3 should be the same, eventually xapi-storage and xcp-idl should be folded together. So lets copy the definition from xapi-storage to xcp-idl for now.

Reply:
That sounds good. But as we've discussed on the other PR, we should consider having 4 lists:
* Attached (with the `sr` references so that we can use it straight away with other calls)
* Existing (created)
* Creatable
* Incomplete

Reply:
I misunderstood this, we want to make xcp-idl look like SMAPIv3

Reply:
I've thought about this, and I think it would be nice to explain that when `sr` is not `None`, this field must be `false`, because even though the configuration is complete, the SR already exists, so we cannot create it. (Or we could rename this to something more explicit like `creatable` / `can_create` / .... .)

Reply:
@edwintorok You've actually anwsered my question in https://github.com/xapi-project/xapi-storage-script/pull/58/commits/cb3e6fcec55fc02fb3b2d2359aede697966fba7f : 
```
'complete' simply means there is no need for further probing,
it doesn't mean its suitable only for SR.create.
Both SR.create and SR.attach configurations are 'complete'.
```
So we should update the description here too.